src/python/grpcio/grpc/__init__.py

```diff
@@ -833,7 +833,11 @@ class HandlerCallDetails(six.with_metaclass(abc.ABCMeta)):
 
 
 class GenericRpcHandler(six.with_metaclass(abc.ABCMeta)):
```

Comment:
Rather than modifying this class in a tricky way I think we should recenter this change around a new class that looks like
```py
class ServiceRpcHandler(six.with_metaclass(abc.ABCMeta)):
  def service_name(self):
    # returns a string like "my_package.my_subpackage.MyService"
  def service_rpc(self, handler_call_details):
    # services an RPC.
```
, under the banner of "if we're going to introduce the concept of a service into the static API, let's do it plainly and simply and conventionally". I'm open to the idea of `ServiceRpcHandler` being a subinterface of `GenericRpcHandler` but don't like the way it would force `ServiceRpcHandler.service` to have the same semantics as `GenericRpcHandler.service`.